### PR TITLE
Feature/version from csproj

### PR DESCRIPTION
**Pull request by jlariono**
Adding support for pulling version number to build from `.csproj` files.

### CODE REVIEW

File: Whiskey/Functions/New-WhiskeySemanticVersion.ps1 [hunk 1]
```diff
@@ -8,11 +8,19 @@ function New-WhiskeySemanticVersion
...
+    If passed a `Version`, it will return that version with build metadata attached. Any build metadata on the passed-in version is replaced. On a build server, build metadata is the build number, source control branch, and commit ID, e.g. `80.master.deadbee`. When run by developers, the build metadata is the current username and computer name, e.g. `whiskey.desktop001`.
 
-    If not passed a version, or the version passed is null or empty, a date-based version number is generated for you. The major number is the year and the minor number is the month and day, e.g. `2017.808`. If run by a developer, the patch number is set to `0`. If run on a build server, the build number is used.
+    Alternatively, you may pass a valid file path to the `Path` parameter and the function will pull the version number from that file.
```
Comment (splatteredbits) [inline Whiskey/Functions/New-WhiskeySemanticVersion.ps1]: Combine these two paragraphs.

File: Whiskey/Functions/New-WhiskeySemanticVersion.ps1 [hunk 2]
```diff
@@ -8,11 +8,19 @@ function New-WhiskeySemanticVersion
...
+    If passed a `Version`, it will return that version with build metadata attached. Any build metadata on the passed-in version is replaced. On a build server, build metadata is the build number, source control branch, and commit ID, e.g. `80.master.deadbee`. When run by developers, the build metadata is the current username and computer name, e.g. `whiskey.desktop001`.
 
-    If not passed a version, or the version passed is null or empty, a date-based version number is generated for you. The major number is the year and the minor number is the month and day, e.g. `2017.808`. If run by a developer, the patch number is set to `0`. If run on a build server, the build number is used.
+    Alternatively, you may pass a valid file path to the `Path` parameter and the function will pull the version number from that file.
 
-    Pass any prerelease metadata to the `Prerelease` parameter. If `Version` has a value, then the `Prerelease` parameter is ignored.
+    The `Path` parameter currently supports the following files:
+
+    * Node.js `package.json` files.
```
Comment (splatteredbits) [inline Whiskey/Functions/New-WhiskeySemanticVersion.ps1]: Add more details about where in each file the version is pulled from.

**Comment by jlariono** [comment]
@splatteredbits Made requested changes to `New-WhiskeySemanticVersion` docs. Ready for re-review.

File: Whiskey/Functions/New-WhiskeySemanticVersion.ps1 [hunk 3]
```diff
@@ -8,15 +8,28 @@ function New-WhiskeySemanticVersion
...
-    Alternatively, you may pass a valid file path to the `Path` parameter and the function will pull the version number from that file.
+    If passed a `Version`, it will return that version with build metadata attached. Alternatively, you may pass a valid file path to the `Path` parameter and the function will pull the version number from that file. Any build metadata on the passed-in version is replaced. On a build server, build metadata is the build number, source control branch, and commit ID, e.g. `80.master.deadbee`. When run by developers, the build metadata is the current username and computer name, e.g. `whiskey.desktop001`.
 
     The `Path` parameter currently supports the following files:
 
-    * Node.js `package.json` files.
-    * PowerShell module manifest files with the file extension `.psd1`.
-    * .NET csproj files with the file extension `.csproj`.
+    * Node.js `package.json` files (JSON key: `version`).
+        ```
```
Comment (splatteredbits) [inline Whiskey/Functions/New-WhiskeySemanticVersion.ps1]: Remove the three backticks. That is Github Markdown syntax. Indent by four spaces instead.

Comment (jlariono) [reply Whiskey/Functions/New-WhiskeySemanticVersion.ps1]: Ah that's right. Fixed. Will need re-approval.

**Pull request merged**